demo/b2share_demo/helpers.py

```diff
@@ -418,8 +419,12 @@ def _save_file(url, filename):
     counter = 0
     while not finished:
         try:
-            f.write(u.read())
-            finished = True
+            chunk = u.read(CHUNK_SIZE)
+            if chunk:
+                f.write(chunk)
+            else:
+                finished = True
         except:
```

Comment:
No, this is an infinite loop if the read fails (e.g. url is not accessible). Except must catch a meaningful error, or set finished on true on exception.